projects/observability/src/shared/graphql/request/builders/specification/specification-builder.ts

```diff
@@ -10,8 +11,31 @@ export class SpecificationBuilder {
   protected readonly selectionBuilder: GraphQlSelectionBuilder = new GraphQlSelectionBuilder();
   protected readonly traceStatusSpecBuilder: TraceStatusSpecificationBuilder = new TraceStatusSpecificationBuilder();
 
-  public buildCompositeSpecification(specifications: Specification[], orderByKey: string): CompositeSpecification {
-    const name = specifications.map(specification => specification.resultAlias()).join('_');
+  /**
+   * If specifications are of type Specification[] then extractFromServerData will return unknown[]
+   * If specifications are of type Dictionary<Specification> then extractFromServerData will return Dictionary<unknown> with the exact mappings
+   * TODO: Add Typings
+   */
+  public buildCompositeSpecification(
+    specifications: Specification[] | Dictionary<Specification>,
+    orderByKey: string
+  ): CompositeSpecification {
+    if (!Array.isArray(specifications)) {
+      const specs = Object.values(specifications);
+      const alias = this.buildSpecName(specs);
+
+      return {
+        resultAlias: () => alias,
+        name: alias,
+        asGraphQlSelections: () => this.selectionBuilder.fromSpecifications(specs),
+        extractFromServerData: serverData => this.extractForSpecificationsObject(serverData, specifications),
+        asGraphQlOrderByFragment: () => ({
+          expression: { key: orderByKey }
+        })
+      };
```

Comment:
would be good to define the separate flows (array vs object extraction) in independent methods.

Reply:
We thought about the same, but discussed and chose the same path
cc: @anandtiwary